### PR TITLE
Left panel

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

**Comment by iqmulator** [comment]
search panel and new chat button
image
thin navbar icons
image
new menu style
image
update message menu
image
pinned/unpinned chats panel
image
new actual chats panel with toggles
image
image
left panel with pinned/unpinned/actual chats and scrollbar
image
update account button menu
image

File: src/dotnet/Chat.UI.Blazor/Components/ChatListNavbarWidget.razor [hunk 1]
```diff
@@ -9,62 +9,116 @@
         + (mustPlayPinned ? "" : " opacity-50")
         + (isPlaying ? " on" : mustPlayPinned ? "" : " off");
 
-    RenderFragment RenderItem(Chat c) =>
+    RenderFragment RenderUnpinned(Chat c) =>
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/ChatListNavbarWidget.razor:12]: Component is quite big. May be it make sense to split it in several components?

File: src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/chat-message-editor.ts [hunk 2]
```diff
@@ -132,6 +132,16 @@ export class ChatMessageEditor {
         this.changeMode();
     }
 
+    public clearAttachments() {
+        const attachments = this.attachments;
+        this.attachments.clear();
+        attachments.forEach(a => {
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/chat-message-editor.ts:138]: Attachments will be empty after calling clear method, URL.revokeObjectURL will never be invoked.

File: src/dotnet/Chat.UI.Blazor/Components/ChatPlaybackToggle.razor [hunk 3]
```diff
@@ -25,13 +27,32 @@
         var isPlayingActive = await ChatUI.IsPlaying.Use(cancellationToken).ConfigureAwait(false);
         var chatPlaybackState = await ChatPlayers.ChatPlaybackState.Use(cancellationToken).ConfigureAwait(false);
         var isPlayingHistorical = chatPlaybackState is HistoricalChatPlaybackState;
+        if (!isPlayingActive)
+            return new Model() {
+                IsPlayingActive = isPlayingActive,
+                IsPlayingHistorical = isPlayingHistorical,
+            };
+        var listenedChatIds = await ChatUI.ListenedChatIds.Use(cancellationToken).ConfigureAwait(false);
+        ChatUI.ListenedChatIds.Value = AddToListened(listenedChatIds);
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/ChatPlaybackToggle.razor:36]: Logic to manage listened chats list should not be here. May be better to put it in ChatUI service.
Consult with me or @alexyakunin

File: src/dotnet/UI.Blazor/Components/Navbar/NavbarGroupToggle.razor [hunk 4]
```diff
@@ -6,26 +6,31 @@
...
-<ButtonSquare Click="@OnClicked" Title="@Title" Class="@btnClass">
-    <svg xmlns="http://www.w3.org/2000/svg" class="navbar-group-icon fill-current h-@IconHeight w-@IconWidth" viewBox="@ViewBox">
-        <path d="@IconPath"></path>
-    </svg>
-</ButtonSquare>
+<div class="flex-x">
+    <div class="h-8 w-1 mr-2 self-center rounded-lg @lineClass">
+    </div>
+    <ButtonSquare Click="@OnClicked" Title="@Title" Class="@btnClass">
+        <svg xmlns="http://www.w3.org/2000/svg" class="navbar-group-icon fill-current h-@IconHeight w-@IconWidth" viewBox="@ViewBox">
```
Comment (crui3er) [inline src/dotnet/UI.Blazor/Components/Navbar/NavbarGroupToggle.razor:16]: NOTE: Tailwind will not include generated classs with expression: h-@IconHeight w-@IconWidth
Do we need arbitrary sizes here?

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/ChatListListenedRemove.razor [hunk 5]
```diff
@@ -0,0 +1,28 @@
+@namespace ActualChat.Chat.UI.Blazor.Components
+@inherits ComputedStateComponent<ImmutableArray<Symbol>>
+@{
+    var cls = "pin-toggle";
+}
+
+<ButtonRound Class="@cls" Click="@OnClick" Title="Remove from Listened">
+    <svg class="fill-current w-6 h-6" xmlns="http://www.w3.org/2000/svg" viewBox="2 1 22 22">
```
Comment (alexyakunin) [inline src/dotnet/Chat.UI.Blazor/Components/ChatListListenedRemove.razor:8]: Not for this iteration, but IMO we should start creating our own icon lib - with a common base class for every icon w/ standard size settings.

Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Components/ChatListListenedRemove.razor:8]: It's a bit annying when all these huge blocks are everywhere around.

File: src/dotnet/Chat.UI.Blazor/Components/ChatListNavbarWidget.razor (hunk 1 above)
Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Components/ChatListNavbarWidget.razor:12]: Agree, makes sense.

File: src/dotnet/Chat.UI.Blazor/Components/ChatPlaybackToggle.razor (hunk 3 above)
Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Components/ChatPlaybackToggle.razor:36]: Yep, note the right place, but also you shouldn't normally modify the state from Compute Methods.

File: src/dotnet/Chat.UI.Blazor/Components/ChatPlaybackToggle.razor [hunk 6]
```diff
@@ -25,13 +27,32 @@
...
 
         return new Model() {
             IsPlayingActive = isPlayingActive,
             IsPlayingHistorical = isPlayingHistorical,
         };
     }
 
+    private ImmutableArray<Symbol> AddToListened(ImmutableArray<Symbol> lastListened) {
```
Comment (alexyakunin) [inline src/dotnet/Chat.UI.Blazor/Components/ChatPlaybackToggle.razor:44]: `AddToListeningChatIds` + move it to `ChatUI`. We'll figure out how to do it better later.

File: src/dotnet/UI.Blazor/Components/AccountDropdown/SignedInMenu.razor [hunk 7]
```diff
@@ -1,24 +1,38 @@
 ﻿@namespace ActualChat.UI.Blazor.Components
 @using ActualChat.Users
 @using ActualChat.UI.Blazor.Services
+@using ActualChat.Hosting
+
+@{
+    var version = "v" + (GetType().Assembly.GetInformationalVersion() ?? "n/a").Replace('+', ' ');
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/Components/AccountDropdown/SignedInMenu.razor:7]: It's better to cache it in `private static readonly Version { get; }` & initialize it from type initializer or in-place. Doing this on every render is a waste.

File: src/dotnet/UI.Blazor/Components/Navbar/NavbarGroupToggle.razor (hunk 4 above)
Comment (alexyakunin) [reply src/dotnet/UI.Blazor/Components/Navbar/NavbarGroupToggle.razor:16]: See `SquareSize` and `SquareSizeExt` types - they do it in a Tailwind-friendly way (+ you can add more sizes there if you want to).

Comment (alexyakunin) [reply src/dotnet/UI.Blazor/Components/Navbar/NavbarGroupToggle.razor:16]: Curious how @crui3er even noticed this :)

**Review by alexyakunin — CHANGES_REQUESTED** [review]
Minor stuff, but still

**Pull request merged**